File: src/csharp/pulse/cdm/scenario/SEScenarioExec.cs
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

namespace Pulse.CDM
{
  public class SEScenarioExec
  {
    protected eSwitch log_to_console;
    protected string data_root_directory;
    protected string output_root_directory;
    protected eSwitch organize_output_directory;

    protected eSwitch auto_serialize_after_actions;
    protected double auto_serialize_period_s;
    protected eSwitch time_stamp_serialized_states;

    protected string engine_configuration_content;
    protected string engine_configuration_filename;

    protected string scenario_content;
    protected string scenario_filename;
    protected string scenario_directory;

    protected string data_request_files_search;

    protected eSerializationFormat contentFormat;
    protected int thread_count;

    public SEScenarioExec()
    {
      Clear();
    }

    public virtual void Clear()
    {
      log_to_console = eSwitch.On;
      data_root_directory = "./";
      output_root_directory = "";
      organize_output_directory = eSwitch.Off;

      auto_serialize_after_actions = eSwitch.Off;
      auto_serialize_period_s = 0;
      time_stamp_serialized_states = eSwitch.On;

      engine_configuration_content = "";
      engine_configuration_filename = "";

      scenario_content = "";
      scenario_filename = "";
      scenario_directory = "";

      data_request_files_search = "";

      contentFormat = eSerializationFormat.JSON;
      thread_count = -1;
    }

    public eSwitch GetLogToConsole() { return log_to_console; }
    public void SetLogToConsole(eSwitch s) { log_to_console = s; }

    public string GetDataRootDirectory() { return data_root_directory; }
    public void SetDataRootDirectory(string s) { data_root_directory = s; }

    public string GetOutputRootDirectory() { return output_root_directory; }
    public void SetOutputRootDirectory(string s) { output_root_directory = s; }
    public eSwitch GetOrganizeOutputDirectory() { return organize_output_directory; }
    public void SetOrganizeOutputDirectory(eSwitch s) { organize_output_directory = s; }

    public eSwitch GetAutoSerializeAfterActions() { return auto_serialize_after_actions; }
    public void SetAutoSerializeAfterActions(eSwitch s) { auto_serialize_after_actions = s; }

    public double GetAutoSerializePeriod_s() { return auto_serialize_period_s; }
    public void SetAutoSerializePeriod_s(double d) { auto_serialize_period_s = d; }

    public eSwitch GetTimeStampSerializedStates() { return time_stamp_serialized_states; }
    public void SetTimeStampSerializedStates(eSwitch s) { time_stamp_serialized_states = s; }

    public string GetEngineConfigurationContent() { return engine_configuration_content; }
    public void SetEngineConfigurationContent(string s) { engine_configuration_content = s; }

    public string GetEngineConfigurationFilename() { return engine_configuration_filename; }
    public void SetEngineConfigurationFilename(string s) { engine_configuration_filename = s; }

    public string GetScenarioContent() { return scenario_content; }
    public void SetScenarioContent(string s)
    {
      scenario_content = s;
      scenario_filename = "";
      scenario_directory = "";
    }

    public string GetScenarioFilename() { return scenario_filename; }
    public void SetScenarioFilename(string s)
    {
      scenario_content = "";
      scenario_filename = s;
      scenario_directory = "";
    }

    public string GetScenarioDirectory() { return scenario_directory; }
    public void SetScenarioDirectory(string s)
    {
      scenario_content = "";
      scenario_filename = "";
      scenario_directory = s;
    }

    public string GetDataRequestFilesSearch() { return data_request_files_search; }
    public void SetDataRequestFilesSearch(string s) { data_request_files_search = s; }

    public eSerializationFormat GetContentFormat() { return contentFormat; }
    public void SetContentFormat(eSerializationFormat s) { contentFormat = s; }

    public int GetThreadCount() { return thread_count; }
    public void SetThreadCount(int c) { thread_count = c; }
  }
}

File: src/java/com/kitware/pulse/cdm/scenario/SEScenarioExec.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

package com.kitware.pulse.cdm.scenario;

import com.google.protobuf.InvalidProtocolBufferException;
import com.google.protobuf.util.JsonFormat;
import com.kitware.pulse.cdm.bind.Enums.eSwitch;
import com.kitware.pulse.cdm.bind.Scenario.ScenarioExecData;
import com.kitware.pulse.cdm.bind.Scenario.eSerializationFormat;
import com.kitware.pulse.utilities.Log;

public class SEScenarioExec
{
  protected eSwitch logToConsole;
  protected String  dataRootDirectory;
  protected String  outputRootDirectory;
  protected eSwitch organizeOutputDirectory;

  protected eSwitch autoSerializeAfterActions;
  protected double  autoSerializePeriod_s;
  protected eSwitch timeStampSerializedStates;

  protected String  engineConfigurationContent;
  protected String  engineConfigurationFilename;

  protected String  scenarioContent;
  protected String  scenarioFilename;
  protected String  scenarioDirectory;

  protected eSerializationFormat  contentFormat;
  protected int threadCount;
  
  public SEScenarioExec()
  {
    clear();
  }
  
  public void clear()
  {
    logToConsole = eSwitch.Off;
    dataRootDirectory = "./";
    outputRootDirectory = "";
    organizeOutputDirectory = eSwitch.Off;
    
    autoSerializeAfterActions = eSwitch.Off;
    autoSerializePeriod_s = 0;
    timeStampSerializedStates = eSwitch.On;
    
    engineConfigurationContent = "";
    engineConfigurationFilename = "";
    
    scenarioContent = "";
    scenarioFilename = "";
    scenarioDirectory = "";
    
    contentFormat = eSerializationFormat.JSON;
    threadCount = -1;
  }
  
  public void copy(SEScenarioExec src)
  {
    SEScenarioExec.load(SEScenarioExec.unload(src), this);
  }
  
  public static void load(ScenarioExecData src, SEScenarioExec dst)
  {
    dst.clear();
    dst.logToConsole = src.getLogToConsole();
    if(src.getDataRootDirectory()!=null)
      dst.dataRootDirectory = src.getDataRootDirectory();
    if(src.getOutputRootDirectory()!=null)
      dst.outputRootDirectory = src.getOutputRootDirectory();
    dst.organizeOutputDirectory = src.getOrganizeOutputDirectory();
    
    dst.autoSerializeAfterActions = src.getAutoSerializeAfterActions();
    dst.autoSerializePeriod_s = src.getAutoSerializePeriodS();
    dst.timeStampSerializedStates = src.getTimeStampSerializedStates();
    
    if(src.getEngineConfigurationContent()!=null)
      dst.engineConfigurationContent = src.getEngineConfigurationContent();
    else if(src.getEngineConfigurationFilename()!=null)
      dst.engineConfigurationFilename = src.getEngineConfigurationFilename();
    
    if(src.getScenarioContent()!=null)
      dst.scenarioContent = src.getScenarioContent();
    else if(src.getScenarioFilename()!=null)
      dst.scenarioFilename = src.getScenarioFilename();
    else if(src.getScenarioDirectory()!=null)
      dst.scenarioDirectory = src.getScenarioDirectory();
    
    dst.contentFormat = src.getContentFormat();
    dst.threadCount = src.getThreadCount();
  }
  public static ScenarioExecData unload(SEScenarioExec src)
  {
    ScenarioExecData.Builder dst = ScenarioExecData.newBuilder();
    unload(src,dst);
    return dst.build();
  }
  protected static void unload(SEScenarioExec src, ScenarioExecData.Builder dst)
  {
    dst.setLogToConsole(src.logToConsole);
    if (!src.dataRootDirectory.isEmpty())
      dst.setDataRootDirectory(src.dataRootDirectory);
    if (!src.outputRootDirectory.isEmpty())
      dst.setOutputRootDirectory(src.outputRootDirectory);
    dst.setOrganizeOutputDirectory(src.organizeOutputDirectory);

    dst.setAutoSerializeAfterActions(src.autoSerializeAfterActions);
    dst.setAutoSerializePeriodS(src.autoSerializePeriod_s);
    dst.setTimeStampSerializedStates(src.timeStampSerializedStates);
    
    if (!src.engineConfigurationContent.isEmpty())
      dst.setEngineConfigurationContent(src.engineConfigurationContent);
    else if (!src.engineConfigurationFilename.isEmpty())
      dst.setEngineConfigurationFilename(src.engineConfigurationFilename);
    
    if (!src.scenarioContent.isEmpty())
      dst.setScenarioContent(src.scenarioContent);
    else if (!src.scenarioFilename.isEmpty())
      dst.setScenarioFilename(src.scenarioFilename);
    else if (!src.scenarioDirectory.isEmpty())
      dst.setScenarioDirectory(src.scenarioDirectory);
    
    dst.setContentFormat(src.contentFormat);
    dst.setThreadCount(src.threadCount);
  }
  
  public String toJSON() 
  {
    String json;
    try
    {
      json = JsonFormat.printer().print(SEScenarioExec.unload(this));
    }
    catch (InvalidProtocolBufferException e)
    {
      Log.error("Unable to generate json from SEScenarioExec");
      json = "";
    }
    return json;
  }
  
  public eSwitch getLogToConsole() { return logToConsole; }
  public void setLogToConsole(eSwitch s) { logToConsole=s; }

  public String getDataRootDirectory() { return dataRootDirectory; }
  public void setDataRootDirectory(String s) { dataRootDirectory=s; }
  
  public String getOutputRootDirectory() { return outputRootDirectory; }
  public void setOutputRootDirectory(String s) { outputRootDirectory=s; }
  
  public eSwitch getOrganizeOutputDirectory() { return organizeOutputDirectory; }
  public void setOrganizeOutputDirectory(eSwitch s) { organizeOutputDirectory=s; }

  public eSwitch getAutoSerializeAfterActions() { return autoSerializeAfterActions; }
  public void setAutoSerializeAfterActions(eSwitch s) { autoSerializeAfterActions=s; }

  public double getAutoSerializePeriod_s() { return autoSerializePeriod_s; }
  public void setAutoSerializePeriod_s(double d) { autoSerializePeriod_s=d; }

  public eSwitch getTimeStampSerializedStates() { return timeStampSerializedStates; }
  public void setTimeStampSerializedStates(eSwitch s) { timeStampSerializedStates=s; }

  public String getEngineConfigurationContent() { return engineConfigurationContent; }
  public void setEngineConfigurationContent(String s) { engineConfigurationContent=s; }

  public String getEngineConfigurationFilename() { return engineConfigurationFilename; }
  public void setEngineConfigurationFilename(String s) { engineConfigurationFilename=s; }

  public String getScenarioContent() { return scenarioContent; }
  public void setScenarioContent(String s) { scenarioContent=s; }

  public String getScenarioFilename() { return scenarioFilename; }
  public void setScenarioFilename(String s) { scenarioFilename=s; }

  public String getScenarioDirectory() { return scenarioDirectory; }
  public void setScenarioDirectory(String s) { scenarioDirectory=s; }

  public eSerializationFormat getContentFormat() { return contentFormat; }
  public void setContentFormat(eSerializationFormat s) { contentFormat=s; }
  
  public int getThreadCount() { return threadCount; }
  public void setThreadCount(int c) { threadCount=c; }
  
}
